modules/post/multi/manage/screenshare.rb

```diff
@@ -21,7 +21,17 @@ def initialize(info = {})
         'Author' => [ 'timwr'],
         'Platform' => [ 'linux', 'win', 'osx' ],
         'SessionTypes' => [ 'meterpreter' ],
-        'DefaultOptions' => { 'SRVHOST' => '127.0.0.1' }
+        'DefaultOptions' => { 'SRVHOST' => '127.0.0.1' },
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              espia_image_get_dev_screen
```

Comment:
Kind of an awkward case here, but AFAIK the espia extension is only available on Windows but this module support Linux, Windows and OSX. Including this as a required command will cause it to always be marked as incompatible on Linux and OS X because we can't mark it as only requiring espia when on Windows (which would eliminate say the PHP, Java and Python Meterpreters on Windows).

I don't think we have a precedence set for how to handle cases where a module is capable of identifying the sessions capabilities at runtime and working around them. That's not quite what this one does though since it just checks the platform and loads espia on L64 but it could be updated to check prioritize espia over the stdapi version when it's available.

Reply:
FYI if I remember correctly the espia extension is only used to workaround a bug where `meterpreter > screenshot` leaks memory on Windows. You could probably just leave that line out here.

Reply:
> this module support[s] Linux, Windows and OSX

Looks like mettle doesn't support the screenshot capabilities:
```
msf6 payload(linux/x64/meterpreter_reverse_http) > use multi/manage/screenshare
msf6 post(multi/manage/screenshare) > run session=-1

[!] SESSION may not be compatible with this module:
[!]  * missing Meterpreter features: stdapi_ui_desktop_screenshot, stdapi_ui_send_keyevent, stdapi_ui_send_mouse, espia_image_get_dev_screen
[*] Using URL: http://127.0.0.1:8080/97kauX3i13yHsb
[*] Server started.
[*] Sent screenshare html to 127.0.0.1
[-] Exception handling request: stdapi_ui_desktop_screenshot: Operation failed: The command is not supported by this Meterpreter type (x64/linux)
```

And for OSX the module works, but gives a warning for the espia extension:
```
msf6 post(multi/manage/screenshare) > run session=-1

[!] SESSION may not be compatible with this module:
[!]  * unloadable Meterpreter extension: espia
[*] Using URL: http://127.0.0.1:8080/TfUEwpfy0ll
[*] Server started.

[*] Sent screenshare html to 127.0.0.1
```

There's not many modules that use the espia command. I can update the session validation logic to ignore espia command names/extensions if the target != windows? I'm not strongly opinionated here.

Reply:
One possible solution could be to remove the requirement for this espia command, and update the code later on when it's used to guard for cases in which it's not available. When espia is not available, since the `stdapi_ui*` family of commands *are* required, the module can failover to that technique.

This basically automates the tribal knowledge that espia is the preferred method when it's available (only on the native Windows Meterpreter).